main_sweeps.py

```diff
@@ -0,0 +1,233 @@
+import argparse
+import json
+import os
+import shutil
+from typing import Any, Dict, Type
+
+import numpy as np
+import pandas as pd
+from pyntcloud import PyntCloud
+from pyquaternion import Quaternion
+
+from argoverse.utils.se3 import SE3
+from argoverse.utils.transform import quat2rotmat
+from main import get_calibration_info, round_to_micros, write_ply
+from nuscenes.nuscenes import NuScenes
+
+"""
+Converts the nuScenes unannotated sweeps into Argoverse format.
+
+In the nuScenes dataset, samples contain annotated data sampled at a frequency
+of 2Hz, whereas sweeps contains all the unannotated data. The capture frequency
+for camera is 12Hz and for LiDar it 20Hz. We also convert samples to unannotated 
+Argoverse format.
+
+"""
+
+# nuscenes sensor on the left, corresponding argoverse sensor on right
+SENSOR_NAMES = {
+    "LIDAR_TOP": "lidar",
+    "CAM_FRONT": "ring_front_center",
+    "CAM_FRONT_LEFT": "ring_front_left",
+    "CAM_FRONT_RIGHT": "ring_front_right",
+    "CAM_BACK_LEFT": "ring_side_left",
+    "CAM_BACK_RIGHT": "ring_side_right",
+}
+
+# 3-letter abbreviation of nuScene city names
+CITY_TO_ID = {
+    "singapore-onenorth": "SON",
+    "boston-seaport": "BSP",
+    "singapore-queenstown": "SQT",
+    "singapore-hollandvillage": "SHV",
+}
+
+
+def main(args: argparse.Namespace) -> None:
+    """
+    Convert sweeps and samples into (unannotated) Argoverse format. Overview of algorithm:
+
+    1) Iterate over all scenes in the NuScenes dataset. For each scene, obtain first sample in the scene.
+
```

Comment:
i think newlines in between can be removed

Reply:
done